system_test/aest_hard_fork_SUITE.erl

```diff
@@ -642,6 +691,196 @@ new_node_can_mine_spend_tx_on_old_chain_using_new_protocol(Cfg) ->
+    %% Check protocol version of the block with mined transacion on all nodes.
+    %% Cannot smoke test that block with contract tx mined by new node with old protocol is synced by old_node1.
+    check_protocol_version_on_nodes([new_node3, new_node4], OldProtocolVersion,
+                                    ContractCallBlockHash, ContractCallBlockHeight, Cfg),
+    %% Create contract call compute transaction.
+    {ContractCallCompute, DecContractCallCompute} =
+        make_contract_tx(contract_call_compute_tx, PubKey3, DecPubKey3, ContractPubKey, Cfg),
+    {ok, Nonce2} = get_next_nonce(new_node3, DecPubKey3, Cfg),
+    #{tx := ContractCallComputeTx} =
+        get_contract_tx(new_node3, 'PostContractCallCompute', fun aect_call_tx:new/1,
+                        Nonce2, ContractCallCompute, DecContractCallCompute, Cfg),
+    %% Post contract call compute transaction and wait until it's mined.
+    #{block_hash := ContractCallComputeBlockHash, block_height := ContractCallComputeBlockHeight} =
+        post_contract_tx_and_wait_in_chain(new_node3, PubKey3, Nonce2, ContractPubKey, contract_call_compute_tx,
+                                           ContractCallComputeTx, Cfg),
+    %% Check protocol version of the block with mined transaction on all nodes.
+    %% Cannot smoke test that block with contract tx mined by new node with old protocol is synced by old_node1.
+    check_protocol_version_on_nodes([new_node3, new_node4], OldProtocolVersion,
+                                    ContractCallComputeBlockHash, ContractCallComputeBlockHeight, Cfg),
+    ok.
+
+%% New node can sync the old chain from old node and can start mining
+%% on the top of the old chain. The new node can mine blocks using the
+%% new protocol and include contract create, contract call and contract
+%% call compute transactions in the blocks. The blocks with the
+%% contract transactions can be synced among other new nodes.
+new_node_can_mine_contract_on_old_chain_using_new_protocol(Cfg) ->
+    {_, {_Saver, SavedCfg}} = proplists:lookup(saved_config, Cfg),
+    {_, old_node1} = proplists:lookup(old_node_left_running_with_old_chain, SavedCfg),
+    {_, new_node3} = proplists:lookup(new_node_left_running_with_synced_old_chain, SavedCfg),
+    {_, TopHeight} = proplists:lookup(db_backup_top_height, Cfg),
+    {_, _TopHash} = proplists:lookup(db_backup_top_hash, Cfg),
+    OldProtocolVersion = old_protocol_version(),
+    NewProtocolVersion = new_protocol_version(),
+    HeightOfNewProtocol = ?HEIGHT_OF_NEW_PROTOCOL(TopHeight),
+    start_node(new_node4, Cfg),
+    wait_for_height_syncing(TopHeight, [new_node4], {{100000, ms}, {1000, blocks}}, Cfg),
+    ok = mock_pow_on_node(new_node3, Cfg),
+    ok = mock_pow_on_node(new_node4, Cfg),
+    %% Kill old node - not needed.
+    aest_nodes:kill_node(old_node1, Cfg),
+    %% expected_mine_rate is lowered so contract gets mined faster.
+    ErlCmd = make_erl_cmd("application:set_env(aecore, expected_mine_rate, ~p)", [10]),
```

Comment:
I think this should not be needed. System tests are run with a specific mining rate as part of the docker setup.
see aest_nodes for the setup interface.